.github/workflows/ci.yml

```diff
@@ -0,0 +1,31 @@
+# https://help.github.com/en/categories/automating-your-workflow-with-github-actions
+
+on:
+  - pull_request
+  - push
+
+name: CI
+
+jobs:
+  php-code-sniffer:
+    name: PHP Code Sniffer
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: "Cache dependencies installed with composer"
+        uses: actions/cache@v1
+        with:
+          path: ~/.composer/cache
+          key: php${{ matrix.php-version }}-composer-${{ hashFiles('**/composer.json') }}
```

Comment:
`matrix.php-version` seems to be resolving to an empty string, from the logs, e.g. https://github.com/MyIntervals/emogrifier/runs/372326499

Also, an [example I found](https://github.com/actions/cache/blob/master/examples.md#php---composer) is using `composer.lock` for the hash rather than `composer.json`.

Reply:
This jobs does not use a PHP version matrix. So adding it PHP matrix version to the cache key indeed did not make any sense. (It will do, though, for jobs with a version matrix, e.g., the PHP linting or the unit tests.) Good catch!

Reply:
We cannot hash `composer.lock` because we don't have one, and it gets generated only in the next step (`composer install`).

Reply:
I have added the explicit PHP version using a matrix now.

Reply:
I'll look into the `restore-keys` to find out what's the difference to `key`.

Reply:
This seems to be the official way AFAICT.